pkg/providers/cloudstack/validator.go

```diff
@@ -105,6 +97,28 @@ func (v *Validator) ValidateCloudStackDatacenterConfig(ctx context.Context, data
 	return nil
 }
 
+func (v *Validator) validateDomainAndAccount(ctx context.Context, datacenterConfig *anywherev1.CloudStackDatacenterConfig) error {
+	if (datacenterConfig.Spec.Domain != "" && datacenterConfig.Spec.Account == "") ||
+		(datacenterConfig.Spec.Domain == "" && datacenterConfig.Spec.Account != "") {
+		return fmt.Errorf("both domain and account must be specified or none of them must be specified")
```

Comment:
i think you can specify domain without account

Reply:
> i think you can specify domain without account

When deploy a VM in a sub-domain if a matching account is not passed along with the domain ID the VM is created under the parent domain. This is why we enforce both account and domain.